packages/dds/tree/src/core/schema-stored/storedSchemaRepository.ts

```diff
@@ -59,7 +62,8 @@ export interface MutableTreeStoredSchema extends TreeStoredSchemaSubscription {
 export class TreeStoredSchemaRepository implements MutableTreeStoredSchema {
 	protected nodeSchemaData: BTree<TreeNodeSchemaIdentifier, TreeNodeStoredSchema>;
 	protected rootFieldSchemaData: TreeFieldStoredSchema;
-	protected readonly events = createEmitter<SchemaEvents>();
+	protected readonly _events = createEmitter<SchemaEvents>();
+	public readonly events: Listenable<SchemaEvents> = this._events;
```

Comment:
What's the motivation for having `_events`?

Reply:
Is that to keep the ability to fire events encapsulated?

Reply:
Yes, exactly. Usually it's `#events`, but in this case it's protected so we use `_events`.

Reply:
In the future I'd like to explore if there is some clever way to have it work without having to define multiple fields, but I'm not sure if that's possible.

Reply:
Yeah, I was wondering the same thing.
Having usages of the class be indirected through an interface would take care of it.

Otherwise, I would find this pattern less ugly if we had one public member for listerning only and one private member for firing only.
You'd then initialize them in the ctor by doing `{this.events, this.eventEmitter} = createEmitter<SchemaEvents>();` or something like that.

Reply:
That's interesting! That could work. Although, I don't think it's possible to do the destructuring in one line like that - you'd have to use a local and at least three lines. But the splitting of the two does make sense.

Another option would be to do `public readonly events = createEmitter()` and have it already be the "listen-only" interface, but there's a static helper `emit(this.events, key, ...)` which does a cast inside to get the emitting version. That would be the least amount of boilerplate, but is kind of unconventional and weird (even if we ensured the cast was safe by e.g. using a listen-only class instead of merely a listen-only interface).

Reply:
Nit: `eventEmitter` might be a better property name for these to help clarify the difference